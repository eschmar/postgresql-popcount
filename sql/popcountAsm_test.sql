CREATE EXTENSION popcount;

SELECT popcountAsm(15::bit(4));
SELECT popcountAsm(7::bit(4));

SELECT popcountAsm(15::bit(8));
SELECT popcountAsm(7::bit(8));

SELECT popcountAsm(15::bit(12));
SELECT popcountAsm(7::bit(12));

SELECT popcountAsm(15::bit(32));
SELECT popcountAsm(7::bit(32));

SELECT popcountAsm(30::bit(32));
SELECT popcountAsm(127::bit(32));

SELECT sum(popcountAsm(x::bit(512))) from generate_series(1,1000000) x;
SELECT sum(popcountAsm(x::bit(500))) from generate_series(1,1000000) x;
