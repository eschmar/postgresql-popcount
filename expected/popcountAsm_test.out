CREATE EXTENSION popcount;
ERROR:  extension "popcount" already exists
SELECT popcountAsm(15::bit(4));
 popcountasm 
-------------
           4
(1 row)

SELECT popcountAsm(7::bit(4));
 popcountasm 
-------------
           3
(1 row)

SELECT popcountAsm(15::bit(8));
 popcountasm 
-------------
           4
(1 row)

SELECT popcountAsm(7::bit(8));
 popcountasm 
-------------
           3
(1 row)

SELECT popcountAsm(15::bit(12));
 popcountasm 
-------------
           4
(1 row)

SELECT popcountAsm(7::bit(12));
 popcountasm 
-------------
           3
(1 row)

SELECT popcountAsm(15::bit(32));
 popcountasm 
-------------
           4
(1 row)

SELECT popcountAsm(7::bit(32));
 popcountasm 
-------------
           3
(1 row)

SELECT popcountAsm(30::bit(32));
 popcountasm 
-------------
           4
(1 row)

SELECT popcountAsm(127::bit(32));
 popcountasm 
-------------
           7
(1 row)

SELECT sum(popcountAsm(x::bit(512))) from generate_series(1,1000000) x;
   sum   
---------
 9884999
(1 row)

SELECT sum(popcountAsm(x::bit(500))) from generate_series(1,1000000) x;
   sum   
---------
 9884999
(1 row)

